packages/webapp/components/invite/common.ts

```diff
@@ -1,9 +1,29 @@
 import { ReferralCampaignKey } from '@dailydotdev/shared/src/hooks';
 import { Author } from '@dailydotdev/shared/src/graphql/comments';
+import { ActionType } from '@dailydotdev/shared/src/graphql/actions';
+import { Feature } from '@dailydotdev/shared/src/lib/featureManagement';
+import { JSONValue } from '@growthbook/growthbook';
+import { AuthTriggers } from '@dailydotdev/shared/src/lib/auth';
 
 export interface JoinPageProps {
-  token?: string;
-  redirectTo?: string;
+  token: string;
   referringUser: Author;
   campaign: ReferralCampaignKey;
 }
+
+export interface InviteComponentProps {
+  referringUser: Author;
+  campaign: ReferralCampaignKey;
+  handleAcceptClick: () => void | Promise<void>;
+  isLoading: boolean;
+  isSuccess: boolean;
+  redirectTo: string;
+}
+
+export interface ComponentConfig {
+  actionType: ActionType;
+  feature?: Feature<JSONValue>;
```

Comment:
```suggestion
  feature?: Feature;
```
This would work right?

Reply:
Nope, it requires an argument. That's why the dependency above:
`    "@growthbook/growthbook": "https://gitpkg.now.sh/dailydotdev/growthbook/packages/sdk-js?a0d1a3add257acdbf5353dfd690547feebbada92",
`

Reply:
so did we need to fork the growthbook library? Sry if I'm missing some context, but if that is the case, why did we need to do this?

Reply:
@nensidosari Let's move these types to shared then?
It will make more sense since that already has GB